docs/development-guidelines/trouble-shooting.md

```diff
@@ -51,6 +51,10 @@ module openAiRoleUser 'core/security/role.bicep' = if (createRoleForUser) {
 
 ```
 
+### Global deployment endpoints related errors
+
+If you are implementing [Global Deployment Type](https://learn.microsoft.com/azure/ai-services/openai/how-to/deployment-types#global-versus-regional-deployment-types) and you come across `azd down` errors, or other issues, follow [this link](https://github.com/Azure-Samples/azd-template-artifacts/blob/main/docs/development-guidelines/global-deployment.md)
+
```

Comment:
This guidance does not sit well with me at the moment.  It feels like a bug on the cognitive services resource provider. If deleting a resource does not internally delete its own child resources (model deployments in this case) then this feels like a negative user experience and will cause issues not only in `azd down` scenarios but anytime a cognitive services account is deleted.

I'd recommend reporting an issue with the cognitive services team before I'd be comfortable promoting this type of guidance to users.

This is similar to saying if a user created a cosmos database with tables and then you try to delete the database it will fail because it contains tables.

Reply:
This solution is currently preferrable to a failing `azd down` with the consequent cost to customers. I recommend the `azd` evaluating a resolution with the services team. For now the documentation page will stay in place.

Reply:
I also don't agree with the guidance in the documentation and that's why I asked @wbreza for a second opinion in reviewing this PR.

Reply:
@cassiebreviu remember you were planning to check with the AI service team of whether this change was intentional since it wasn't the case earlier? Any feedback here?

Can we use [deployment stacks since that avoids this issue](https://github.com/Azure/azure-dev/issues/4210#issuecomment-2307476286) and we will not need the workaround via hooks? cc @wbreza @kristenwomack @anfibiacreativa @cassiebreviu @pamelafox

Reply:
@rajeshkamal5050 - The Cognitive Service account and child deployments are properly deleted when using the new deployment stacks integration.

Deployment stacks can be enabled by running `azd config set alpha.deployment.stacks on` command and re-provisioning your application.

Reply:
Please @wbreza @kristenwomack feel free to update the documentation directly or suggest changes in the PR itself pointing to how the new solution is integrated after testing it on the https://github.com/Azure-Samples/openai-chat-vision-quickstart. If the validation passes, we can discuss it as the way to go. For now the pre-up needs to stay in place as an evident short term recommendation, for the pipeline not to throw an error on `azd down`, at least while we're refactoring the output according to the Template Lifecycle schema.